src/XrdOssStats/XrdStatsConfig.cc

```diff
@@ -0,0 +1,128 @@
+        result = std::chrono::steady_clock::duration(0);
+        return true;
+    }
+    std::chrono::steady_clock::duration dur(0);
+    auto strValue = duration;
+    while (!strValue.empty()) {
+        std::size_t pos;
+        double value;
+        try {
+            value = std::stod(strValue, &pos);
+        } catch (std::invalid_argument const &exc) {
+            errmsg = "Invalid number provided as timeout: " + strValue;
+            return false;
+        } catch (std::out_of_range const &exc) {
+            errmsg = "Provided timeout out of representable range: " + std::string(exc.what());
+            return false;
+        }
+        if (value < 0) {
+            errmsg = "Provided timeout was negative";
+            return false;
+        }
+        strValue = strValue.substr(pos);
+        char unit[3] = {'\0', '\0', '\0'};
+        if (!strValue.empty()) {
+            unit[0] = strValue[0];
+            if (unit[0] >= '0' && unit[0] <= '9') {unit[0] = '\0';}
+        }
+        if (strValue.size() > 1) {
+            unit[1] = strValue[1];
+            if (unit[1] >= '0' && unit[1] <= '9') {unit[1] = '\0';}
+        }
+        if (!strncmp(unit, "ns", 2)) {
+            dur += std::chrono::duration_cast<typeof(dur)>(std::chrono::duration<double, std::nano>(value));
+        } else if (!strncmp(unit, "us", 2)) {
+            dur += std::chrono::duration_cast<typeof(dur)>(std::chrono::duration<double, std::micro>(value));
+        } else if (!strncmp(unit, "ms", 2)) {
+            dur += std::chrono::duration_cast<typeof(dur)>(std::chrono::duration<double, std::milli>(value));
+        } else if (!strncmp(unit, "s", 1)) {
+            dur += std::chrono::duration_cast<typeof(dur)>(std::chrono::duration<double>(value));
+        } else if (!strncmp(unit, "m", 1)) {
+            dur += std::chrono::duration_cast<typeof(dur)>(std::chrono::duration<double, std::ratio<60>>(value));
+        } else if (!strncmp(unit, "h", 1)) {
+            dur += std::chrono::duration_cast<typeof(dur)>(std::chrono::duration<double, std::ratio<3600>>(value));
+        } else if (strlen(unit) > 0) {
+            errmsg = "Unknown unit in duration: " + std::string(unit);
+            return false;
+        } else {
+            errmsg = "Unit missing from duration: " + duration;
+            return false;
+        }
+        strValue = strValue.substr(strlen(unit));
+    }
+    result = dur;
+    return true;
+}
+
+///
+// The following functions export the plugin to the
+// XRootD framework
+
+extern "C" {
+
+XrdOss *XrdOssAddStorageSystem2(XrdOss       *curr_oss,
+                                XrdSysLogger *logger,
+                                const char   *config_fn,
+                                const char   *parms,
+                                XrdOucEnv    *envP)
+{
+                   
+    XrdSysError log(logger, "fsstats_");
+    try {
+        return new StatsFileSystem(curr_oss, logger, config_fn, envP);
```

Comment:
This where you would hand off curr_oss to the wrapper.

Reply:
I believe this is taken care of in the latest push (though I think the comment is wrong, no?  It's handed off to the base class in the constructor of `StatsFileSystem`).